### ISSUE TITLE
Enhance Should Contains for bytes

### ISSUE THREAD

**Issue by WisniewskiP** [robotframework/robotframework#5054]
To use `Should Contains` with bytes, item that is searched needs to be first converted to bytes.
```robot
Test
    ${binaryVar} =    Convert To Bytes    \x00\x01\x02\xA0\xB0
    Should Contain    ${binary1Var}    \xA0
```
result:
```
TypeError: a bytes-like object is required, not 'str'
```

for int the value is converted automatically:
```robot
Test Should Contain int
    ${binary1Var} =    Convert To Bytes    \x00\x01\x02\xA0\xB0
    ${item} =    Convert To Bytes    \xC0

    Should Contain    ${binary1Var}    ${1000}
```
this fails with:
```
ValueError: byte must be in range(0, 256)
```
this is expected.

**Comment by droeland** [comment]
Can I take this one up?

**Comment by pekkaklarck** [comment]
Thanks for the PR @droeland. It looks pretty good, but there are some smallish things that could be still enhanced. See the PR review for more details.

**Comment by Abhilash-du** [comment]
Hi, I have created a pull request that addresses this issue by enhancing the `should_contain` keyword to handle bytes and improve error handling. The changes include:

- Automatically encoding strings to bytes using 'latin-1' when the container is a bytes object.
- Checking if an integer item is within the range 0-255 before converting it to bytes.
- Raising a `ValueError` with a clear message if the integer is out of range.
- Adding tests to validate the new functionality.

You can view and review the pull request here: [#5176](https://github.com/robotframework/robotframework/pull/5176)

Thank you!

**Comment by pekkaklarck** [comment]
Thanks for the PR @Abhilash-du. Similarly as the earlier #5097 it has some issues that would need to be fixed. As I explained in the comments of your PR, I want this issue closed to get RF 7.1 out soon and I needed to decide which PR to use as the base. I decided to go with the earlier one simply because I had reviewed it in detail earlier and making required changes to it is thus very fast for me. Hopefully you are interested to contribute in the future anyway!
